pkg/operator/rules_test.go

```diff
@@ -246,3 +251,117 @@ func TestGenerateGlobalRules(t *testing.T) {
 		})
 	}
 }
+
+func TestHasRules(t *testing.T) {
+	emptyRules := newFakeClientBuilder().Build()
+	hasRule := newFakeClientBuilder().WithObjects(&monitoringv1.Rules{}).Build()
+	errGettingRules := newFakeClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
+		List: func(_ context.Context, _ client.WithWatch, _ client.ObjectList, _ ...client.ListOption) error {
+			return errors.New("error")
+		}}).Build()
+
+	type test struct {
+		client  client.Client
+		want    bool
+		wantErr bool
+	}
+
+	tests := map[string]test{
+		"no rules": {client: emptyRules, want: false},
+		"has rule": {client: hasRule, want: true},
+		"error":    {client: errGettingRules, want: false, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			r := rulesReconciler{
+				client: tc.client,
+			}
+			got, err := r.hasRules(context.Background())
+			if got != tc.want {
+				t.Errorf("want: %t, got: %t", tc.want, got)
+			}
+			if err != nil && !tc.wantErr {
+				t.Errorf("Unexpected error: %s", err)
+			}
+		})
+	}
+}
+
+func TestScaleRuleConsumers(t *testing.T) {
+	var alertmanagerReplicas int32
```

Comment:
suggestion: Could you initialize this to 0 to make it more clear?

Same with rule-evaluator below.

Reply:
This is an [explicit non-decision in the Go Style Decisions](go/go-style/decisions#non-decisions). Here, I prefer this format because the type is important, and I'm using the zero value - which is well-known and accepted behavior in Go, in my experience.

`alertmanagerReplicas := 0` would be an `int`, not an `int32`.

The alternative is `alertmanagerReplicas := int32(0)` which feels odd to me because it has the format of a type cast while initializing a new variable. If I was initializing a field on a struct to its zero value, I would not assign that a specific value.

The long version (`var alertmanagerReplicas int32 = 0`) [is discouraged](go/go-style/best-practices#vardeclinitialization).

Do you feel strongly about this?

Reply:
I do not - thanks for pointing out those references to me!